group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -4424,14 +4441,113 @@ private ConsumerGroupMember validateConsumerGroupMember(
      * @param context        The request context.
      * @param request        The actual LeaveGroup request.
      *
+     * @return The LeaveGroup response and the records to append.
+     */
+    public CoordinatorResult<LeaveGroupResponseData, CoordinatorRecord> classicGroupLeave(
+        RequestContext context,
+        LeaveGroupRequestData request
+    ) throws UnknownMemberIdException, GroupIdNotFoundException {
+        Group group = groups.get(request.groupId(), Long.MAX_VALUE);
+
+        if (group == null) {
+            throw new UnknownMemberIdException(String.format("Group %s not found.", request.groupId()));
+        }
```

Comment:
i'm guessing we don't use
```
Group group = group(request.memberId());
```
as we throw unknown member id for the old coordinator, is this correct? it seems very counterintuitive unfortunately

Reply:
Yeah, it's counterintuitive..

Reply:
Perhaps, a better strategy overall would be to always use `GroupIdNotFoundException` internally and to translate it to `UnknownMemberIdException` where needed (e.g. by catching and rethrowing) with a comment explaining why.

Reply:
should we hold onto https://github.com/apache/kafka/pull/16073 then?